packages/inference/src/lib/makeRequestOptions.ts

```diff
@@ -54,16 +55,34 @@ export async function makeRequestOptions(
 		throw new Error(`Provider ${provider} requires a model ID to be passed directly.`);
 	}
 
+	const inferenceProviderMapping = providerHelper.clientSideRoutingOnly ?
+		{
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			providerId: removeProviderPrefix(maybeModel!, provider),
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			hfModelId: maybeModel!,
+			status: "live",
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			task: task!
+		} satisfies InferenceProviderModelMapping
+		: await getInferenceProviderMapping({
+			modelId: hfModel,
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			task: task!,
+			provider,
+			accessToken: args.accessToken,
+		}, { fetch: options?.fetch });
+	if (!inferenceProviderMapping) {
+		throw new Error(`We have not been able to find inference provider information for model ${hfModel}.`);
+	}
+
 	const resolvedModel = providerHelper.clientSideRoutingOnly
 		? // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-		  removeProviderPrefix(maybeModel!, provider)
-		: await getProviderModelId({ model: hfModel, provider }, args, {
-				task,
-				fetch: options?.fetch,
-		  });
+		removeProviderPrefix(maybeModel!, provider)
+		: inferenceProviderMapping.providerId;
```

Comment:
```suggestion
const resolvedModel = inferenceProviderMapping.providerId;
```

Since `clientSideRoutingOnly` is check above and `removeProviderPrefix` already applied, I don't think this ternary is necessary anymore.

Reply:
updated in https://github.com/huggingface/huggingface.js/pull/1357